### ISSUE TITLE
Windows Docker Setup: Connection to Ollama "dial tcp 127.0.0.1:11434: connect: connection refused"

### ISSUE THREAD

**Issue by JanWerder** [nilsherzig/LLocalSearch#36]
**Describe the bug**
I'm using Docker Desktop on Windows 11 and followed the instructions in the Readme for the Docker Setup. I'm running Ollama native on Windows.

Docker compose log:
```
docker-compose up
[+] Running 5/5
 ✔ Container llocalsearch-redis-1     Created                                                                      0.0s
 ✔ Container llocalsearch-chromadb-1  Created                                                                      0.0s
 ✔ Container llocalsearch-searxng-1   Created                                                                      0.0s
 ✔ Container llocalsearch-backend-1   Recreated                                                                    0.1s
 ✔ Container llocalsearch-frontend-1  Recreated                                                                    0.1s
Attaching to backend-1, chromadb-1, frontend-1, redis-1, searxng-1
chromadb-1  | Starting 'uvicorn chromadb.app:app' with args: --workers 1 --host 0.0.0.0 --port 8000 --proxy-headers --log-config chromadb/log_config.yml --timeout-keep-alive 30
backend-1   | Server started at http://localhost:8080
backend-1   | 2024/04/04 11:57:02 INFO Starting the server
searxng-1   | SearXNG version 2024.3.29+f5bb64cca
searxng-1   | Use existing /etc/searxng/uwsgi.ini
searxng-1   | Use existing /etc/searxng/settings.yml
searxng-1   | Listen on 0.0.0.0:8080
searxng-1   | [uWSGI] getting INI configuration from /etc/searxng/uwsgi.ini
searxng-1   | [uwsgi-static] added mapping for /static => /usr/local/searxng/searx/static
searxng-1   | *** Starting uWSGI 2.0.23 (64bit) on [Thu Apr  4 11:57:02 2024] ***
searxng-1   | compiled with version: 13.2.1 20231014 on 30 November 2023 14:34:33
searxng-1   | os: Linux-5.15.146.1-microsoft-standard-WSL2 #1 SMP Thu Jan 11 04:09:03 UTC 2024
searxng-1   | nodename: 80f67ff16773
searxng-1   | machine: x86_64
searxng-1   | clock source: unix
searxng-1   | pcre jit disabled
searxng-1   | detected number of CPU cores: 12
searxng-1   | current working directory: /usr/local/searxng
searxng-1   | detected binary path: /usr/sbin/uwsgi
searxng-1   | chdir() to /usr/local/searxng/searx/
searxng-1   | your memory page size is 4096 bytes
searxng-1   | detected max file descriptor number: 1048576
searxng-1   | building mime-types dictionary from file /etc/mime.types...1390 entry found
searxng-1   | lock engine: pthread robust mutexes
searxng-1   | thunder lock: disabled (you can enable it with --thunder-lock)
searxng-1   | uwsgi socket 0 bound to TCP address 0.0.0.0:8080 fd 3
searxng-1   | Python version: 3.11.8 (main, Feb 19 2024, 17:01:17) [GCC 13.2.1 20231014]
searxng-1   | Python main interpreter initialized at 0x7fd776bd92d8
searxng-1   | python threads support enabled
searxng-1   | your server socket listen backlog is limited to 100 connections
searxng-1   | your mercy for graceful operations on workers is 60 seconds
searxng-1   | mapped 1568736 bytes (1531 KB) for 48 cores
searxng-1   | *** Operational MODE: preforking+threaded ***
searxng-1   | added /usr/local/searxng/ to pythonpath.
searxng-1   | spawned uWSGI master process (pid: 7)
searxng-1   | spawned uWSGI worker 1 (pid: 10, cores: 4)
searxng-1   | spawned uWSGI worker 2 (pid: 13, cores: 4)
searxng-1   | spawned 4 offload threads for uWSGI worker 1
searxng-1   | spawned 4 offload threads for uWSGI worker 2
searxng-1   | spawned uWSGI worker 3 (pid: 20, cores: 4)
searxng-1   | spawned 4 offload threads for uWSGI worker 3
searxng-1   | spawned uWSGI worker 4 (pid: 25, cores: 4)
searxng-1   | spawned 4 offload threads for uWSGI worker 4
searxng-1   | spawned uWSGI worker 5 (pid: 30, cores: 4)
searxng-1   | spawned 4 offload threads for uWSGI worker 5
searxng-1   | spawned uWSGI worker 6 (pid: 35, cores: 4)
chromadb-1  | DEBUG:    [04-04-2024 11:57:02] Registering provider: token_config
chromadb-1  | DEBUG:    [04-04-2024 11:57:02] Registering provider: user_token_config
searxng-1   | spawned 4 offload threads for uWSGI worker 6
chromadb-1  | DEBUG:    [04-04-2024 11:57:02] Registering provider: token
searxng-1   | spawned uWSGI worker 7 (pid: 40, cores: 4)
chromadb-1  | DEBUG:    [04-04-2024 11:57:02] Registering provider: token
searxng-1   | spawned uWSGI worker 8 (pid: 42, cores: 4)
searxng-1   | spawned 4 offload threads for uWSGI worker 8
searxng-1   | spawned 4 offload threads for uWSGI worker 7
searxng-1   | spawned uWSGI worker 9 (pid: 50, cores: 4)
searxng-1   | spawned uWSGI worker 10 (pid: 51, cores: 4)
searxng-1   | spawned 4 offload threads for uWSGI worker 9
searxng-1   | spawned uWSGI worker 11 (pid: 56, cores: 4)
searxng-1   | spawned uWSGI worker 12 (pid: 58, cores: 4)
searxng-1   | spawned 4 offload threads for uWSGI worker 12
searxng-1   | spawned 4 offload threads for uWSGI worker 10
searxng-1   | spawned 4 offload threads for uWSGI worker 11
frontend-1  |
frontend-1  | > localsearchui@0.0.1 preview
frontend-1  | > vite preview --host
frontend-1  |
chromadb-1  | WARNING:  [04-04-2024 11:57:03] chroma_server_nofile is set to 65535, but this is less than current soft limit of 1048576. chroma_server_nofile will not be set.
chromadb-1  | INFO:     [04-04-2024 11:57:04] Anonymized telemetry enabled. See                     https://docs.trychroma.com/telemetry for more information.
chromadb-1  | DEBUG:    [04-04-2024 11:57:04] Starting component System
chromadb-1  | DEBUG:    [04-04-2024 11:57:04] Starting component OpenTelemetryClient
chromadb-1  | DEBUG:    [04-04-2024 11:57:04] Starting component SimpleAssignmentPolicy
chromadb-1  | DEBUG:    [04-04-2024 11:57:04] Starting component SqliteDB
chromadb-1  | DEBUG:    [04-04-2024 11:57:04] Starting component QuotaEnforcer
chromadb-1  | DEBUG:    [04-04-2024 11:57:04] Starting component Posthog
chromadb-1  | DEBUG:    [04-04-2024 11:57:04] Starting component LocalSegmentManager
chromadb-1  | DEBUG:    [04-04-2024 11:57:04] Starting component SegmentAPI
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 30 (default app)
chromadb-1  | INFO:     [04-04-2024 11:57:04] Started server process [1]
chromadb-1  | INFO:     [04-04-2024 11:57:04] Waiting for application startup.
chromadb-1  | INFO:     [04-04-2024 11:57:04] Application startup complete.
chromadb-1  | INFO:     [04-04-2024 11:57:04] Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 58 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 13 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 40 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 20 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 42 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 35 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 51 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 10 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 56 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 50 (default app)
searxng-1   | WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x7fd776bd92d8 pid: 25 (default app)
frontend-1  |   ➜  Local:   http://localhost:4173/
frontend-1  |   ➜  Network: http://172.18.0.5:4173/
backend-1   | 2024/04/04 11:57:10 WARN Model does not exist, pulling it model=all-minilm:v2
backend-1   | 2024/04/04 11:57:10 ERROR Model does not exist and could not be pulled model=all-minilm:v2 error="Post \"http://localhost:11434/api/pull\": dial tcp 127.0.0.1:11434: connect: connection refused"
```

I've changed my docker-compose.yaml to localhost for Ollama.
```
services:
  backend:
    image: nilsherzig/llocalsearch-backend:latest
    environment:
      # the ip / url of YOUR Ollama server
      # CHANGE THIS
      - OLLAMA_HOST=http://localhost:11434
```

When I visit the http://localhost:11434/ URL in my browser, the Ollama endpoint comes up. So it seems like the communication between Docker and Ollama isn't working.

How can I resolve that?

**Comment by antback** [comment]
You should put the following to connect from the container to the host.

```
      - OLLAMA_HOST=http://host.docker.internal:11434
```

Regards.

**Comment by nilsherzig** [comment]
I changed the default domains / urls to something more sensible. @antback is right about his solution, but it's upstream by now. Please run "git pull" to get the new configs and try again.

**Comment by JanWerder** [comment]
Perfect, both solutions work, but fixing it in the main branch is of course the best solution. Thanks!

**Comment by nilsherzig** [comment]
nice good to hear :)

**Comment by 0ofta** [comment]
I apologise if there is a better way to communicate this, than reopening this issue, but I got the exact same problem, cloning the master today. 

After some searching, I figured out that **for host.docker.internal to work on Linux** (or at least my Arch Linux install), **docker desktop had to be installed**.

It took me quite some time to figure out, so I wanted to share, in case others have the same trouble. Maybe this could be put in the readme.md.

Thanks for your amazing work!
